### ISSUE TITLE
HandleWindowClick Listener Chrome 74 errors

### ISSUE THREAD

**Issue by jacksonStone** [mlaursen/react-md#829]
After upgrading Chrome to version 74 today began to get very inconsistent failures dealing with HandleWindowClickListener occasionally failing in several places, resulting in a white screen of death. It seems at some point when the event listeners attempt to remove themselves, failures will occur in this area of the code:

```
(0, _handleWindowClickListeners2.default)(this._handleOutsideClick, false);
```
This has happened for us in both the Layover's componentWillUnmount, as well as the DateContainer's ComponentDidUpdate

Exact error thrown is the following:
"TypeError: Cannot convert undefined or null to object"

### Link to a gist or code sample where the issue can be reproduced
Unfortunately reproducing the issue is difficult.  Best pattern we've found is a page with many autocompletes, and selecting items for multiple of them. Eventually the error is generally thrown. Frequently refreshing the page prevents the issue from reproducing as far as we can tell.

### Version
- React - 16.7.0
- React-MD - 1.8.0
- Browser - Chrome 74

**Comment by mlaursen** [comment]
Hmm.. this is a weird one. Does this still happen in the latest version of react-md (1.10.1)?

**Comment by noseglid** [comment]
We're seeing this too on react-md 1.9.3

**Comment by kylewm** [comment]
We have been seeing this error too (on an unrelated, private repo). My best guess right now is that it has to do with calling window.setTimeout (in our case, maybe window.addEventListener in yours) without binding to `window`. I'd be curious if calling 
```js
listener.call(window, 'click', cb)
```
here
https://github.com/mlaursen/react-md/blob/62976901aa42cbe98b5bc5c088baa7f83962b114/src/js/utils/EventUtils/handleWindowClickListeners.js#L28

helps?

**Comment by mlaursen** [comment]
Would you be willing to try that fix locally to see if it works? I'd be more than happy to merge in that fix if it works. I just can't seem to reproduce this yet on Mac or Windows with the few apps I have using react-md

**Comment by jacksonStone** [comment]
I did a bit of digging. I think it has something to do with sentry + reactMD.  Sentry seems to override window methods and use "this" internally in their wrappers.

May explain why you're unable to reproduce. Also based on that I think @kylewm solution will work.  Have not been able to verify locally however due to only our prod builds using sentry

**Comment by jacksonStone** [comment]
Was able to test locally @mlaursen, and was only able to reproduce with sentry enabled. No idea why Chrome 74n surfaced this behavior.

But yeah modifying 

`listener('click', cb);` 

To something more like 

`if(enabled) {
    window.addEventListener('click', cb)
  } else {
    window.removeEventListener('click', cb)
  }`

Fixed the problem for me. Can submit a pull request unless you feel positioned to do so.

**Comment by mlaursen** [comment]
Weird... But yeah, I'll happily accept a PR for it :) Otherwise, I'll add that fix in about an hour when I get off of work. Thanks for testing it out!

**Comment by mlaursen** [comment]
This fix should now be released as 1.10.2.

I also credited you @jacksonStone for the fix [here](https://github.com/mlaursen/react-md/releases/tag/v1.10.2) and [here](https://react-md.mlaursen.com/discover-more/whats-new#v1-10-2-released). Let me know if you'd like different credit or something else.

**Comment by mlaursen** [comment]
@kylewm If this problem still occurs in your app, let me know and I'll reopen this and try another fix.
